src/value.jl

```diff
@@ -179,3 +179,9 @@ DataAPI.defaultarray(::Type{CategoricalValue{T, R}}, N) where {T, R} =
   CategoricalArray{T, N, R}
 DataAPI.defaultarray(::Type{Union{CategoricalValue{T, R}, Missing}}, N) where {T, R} =
   CategoricalArray{Union{T, Missing}, N, R}
+
+# define appropriate handlers for JSON3 interface
+StructTypes.StructType(x::CategoricalValue) = StructTypes.StructType(get(x))
+StructTypes.StructType(::Type{<:CategoricalValue{T}}) where {T} = StructTypes.StructType(T)
+StructTypes.numbertype(::Type{<:CategoricalValue{T}}) where {T <: Number} = T
+(::Type{T})(x::CategoricalValue{<:Number}) where {T <: Number} = T(get(x))
```

Comment:
Sorry, I had missed this method. This PR is really tricky. I'd rather avoid it as many people already complain about the fact that CategoricalArrays defines too many broad methods.

With https://github.com/quinnj/JSON3.jl/pull/72 it isn't needed in my tests.

Reply:
Ah, yeah, good catch! It is tricky: feels like I am playing wack-a-mole!! I had asked about this method **way** at the top of this thread (but it took a less generic form in the first commit), but it would be easy to miss 😉 .

I assume this PR will have to wait until that merges, and ends up in a released version of JSON3?

Reply:
Let's try what @quinnj proposed at https://github.com/quinnj/JSON3.jl/pull/72#issuecomment-683829202:
```suggestion
StructTypes.construct(::Type{T}, x::CategoricalValue{T}) where {T} = T(get(x))
```